eng/scripts/Manage-SDKSuppression-Label.ps1

```diff
@@ -0,0 +1,386 @@
+        [string]$yamlContent
+    )
+
+    try {
+        $yamlParsed = $yamlContent | ConvertFrom-Yaml
+    }
+    catch {
+        return "ConvertFrom-Yaml Error: $($_.Exception.Message)"
+    }
+
+    return $yamlParsed
+}
+
+function Update-Label {
+    param (
+        [Parameter(Mandatory = $true)]
+        $RepoOwner,
+        [Parameter(Mandatory = $true)]
+        $RepoName,
+        [Parameter(Mandatory = $true)]
+        $IssueNumber,
+        [ValidateNotNullOrEmpty()]
+        [Parameter(Mandatory = $true)]
+        $AuthToken,
+        [Parameter(Mandatory)]
+        $sdkNames,
+        [Parameter(Mandatory)]
+        $sdkLabels,
+        [Parameter(Mandatory)]
+        $presentLabels
+    )
+
+    if ($sdkNames.Length -eq 0) {
+        $sdkSuppressionsLabels = $sdkLabels | ForEach-Object { $_.values.'breakingChangeSuppression' }
+        # $sdkSuppressionsLabels | Sort-Object -Unique | ForEach-Object { Write-Output $_ }
+    
+        $removeSdkSuppressionsLabels = [System.Collections.ArrayList]::new()
+        foreach ($prLabel in $presentLabels) {
+            if ($sdkSuppressionsLabels -contains $prLabel) {
+                $sdkSuppressionsApprovedLabel = $null
+                foreach ($sdk in $sdkLabels.GetEnumerator()) {
+                    if ($sdk.Value.'breakingChangeSuppression' -eq $prLabel) {
+                        $sdkSuppressionsApprovedLabel = $sdk.Value.'breakingChangeSuppressionApproved'
+                        break
+                    }
+                }
+                if (-not ($presentLabels -contains $sdkSuppressionsApprovedLabel)) {
+                    Write-Host "updateSdkSuppressionsLabels: PR: $("pr.html_url") To remove the existed suppression label if there are no difference between head commit and base commit for the suppression content."
+                    [void]$removeSdkSuppressionsLabels.Add($prLabel);
+                }
+            }
+        }
+        if ($removeSdkSuppressionsLabels.length -gt 0) {
+            Remove-GithubIssueLabels -RepoOwner $RepoOwner -RepoName $RepoName `
+                -IssueNumber $IssueNumber -LabelsToRemove $willRemoveLabel -AuthToken $token
+            Write-Host "updateSdkSuppressionsLabels: PR: $("pr.html_url") Remove label: $($removeSdkSuppressionsLabels -join ", " )"
+        }
+        else {
+            Write-Host "updateSdkSuppressionsLabels: PR: $("pr.html_url") No Remove label"
+        }
+    }
+    else {
+        $addSdkSuppressionsLabels = [System.Collections.ArrayList]::new()
+        foreach ($sdkName in $sdkNames) {
+            $sdkSuppressionsLabel = $sdkLabels[$sdkName]["breakingChangeSuppression"];
+            if (-not ($presentLabels -contains $sdkSuppressionsLabel)) {
+                [void]$addSdkSuppressionsLabels.Add($sdkSuppressionsLabel);
+            }
+        }
+        if ($addSdkSuppressionsLabels.length -gt 0) {
+            try {
+                Add-GithubIssueLabels -RepoOwner $RepoOwner -RepoName $RepoName `
```

Comment:
If this code needs permissions to write to the repo, it will need a thorough security review, and it will likely need to be split into two actions.  One part that runs on trigger `pull_request` with read-only access to process the business logic, and another part that runs on trigger `workflow_completed` with write access to modify labels.

Reply:
For an example of a work-in-progress, see here:

https://github.com/mikeharder/azure-rest-api-specs/pull/5

Reply:
I saw the example and it was an operation for github label action. If i want to do another script like validate something by Javascript which use 3th-part package. Does the package.json place in eng/scripts/ folder ?